nncf/nncf_network.py

```diff
@@ -371,6 +371,18 @@ def __init__(self, module, input_infos: List[ModelInputInfo] = None,
         self._insertions_into_original_graph = {}  # type: Dict[InsertionPoint, List[Tuple[Callable, OperationPriority]]]
 
 
+        _orig_graph_build_forward_fn = self._get_dummy_forward_fn_for_graph_building(with_input_tracing=True)
+        self._graph_builder = GraphBuilder(_orig_graph_build_forward_fn)
+
+        nncf_wrapped_model = self.get_nncf_wrapped_model()
+        train_only_module_scopes = self.collect_train_only_module_scopes(nncf_wrapped_model, self._graph_builder)
+        if not self.ignored_scopes:
+            self.ignored_scopes = []
+        # No need to compress modules that are executed in train mode only, thus adding them to ignored_scopes to avoid
+        # replacing by nncf modules. As a consequence, model can be compressed in eval mode. No need to care about
+        # spoiling BN statistics, as there're disabled in eval mode.
+        self.ignored_scopes.extend(train_only_module_scopes)
```

Comment:
I think that the `ignored_scopes` attribute, especially if it is public, should be always consistent with what is specified in the config. Perhaps a separate roster of train-only modules should be kept and only concatenated with user's `ignored_scopes` where it is actually necessary to do so, or a separate `_effective_ignored_scopes` attribute should be set up (populated with entries based on `ignored_scopes` and the train module marking) and used internally.

Reply:
Processed separately from `ignored_scope` by `eval_ops_exec_ctx_str`